catalog/cataloger_create_entries.go

```diff
@@ -23,11 +24,23 @@ func (c *cataloger) CreateEntries(ctx context.Context, repository, branch string
 		return nil
 	}
 
-	// validate that we have path on each entry
-	for i := range entries {
-		if !IsNonEmptyString(entries[i].Path) {
+	// validate that we have path on each entry and remember last entry based on path (for dup remove)
+	entriesMap := make(map[string]*Entry, len(entries))
+	for i := len(entries) - 1; i >= 0; i-- {
+		p := entries[i].Path
+		if !IsNonEmptyString(p) {
 			return fmt.Errorf("entry at pos %d, path: %w", i, ErrInvalidValue)
 		}
+		entriesMap[p] = &entries[i]
+	}
+
+	// prepare a list of entries to insert without duplicates
+	entriesToInsert := make([]*Entry, 0, len(entriesMap))
+	for i := range entries {
+		ent := entriesMap[entries[i].Path]
+		if &entries[i] == ent {
+			entriesToInsert = append(entriesToInsert, ent)
+		}
```

Comment:
IIUC the goal here is to keep the insert the *last* object for each key in the order of the *first* appearance of each key.  This is a bit odd.  To keep the files in their expected ordering, I would write something like this (untested, of course):
```go
for i := len(entries)-1, j := len(entries); i >= 0; i-- {
	if &entries[i] == entriesMap[entries[i].Path] {
		j--
		entries[j] := entries[i]
	}
}
entries = entries[j:]
```

It *is* a bit more subtle, but it gives a better ordering I think.  If you don't care about the ordering, just iterate over the map.

If you want the last-last ordering above but without editing the slice in-place:
```go
entriesToInsert := make([]*Entry, len(entriesMap))
for i := len(entries)-1, j := len(entries) - 1; i >= 0; i-- {
	if &entries[i] == entriesMap[entries[i].Path] {
		entriesToInsert[j] := entries[i]
		j--
	}
}
```

Reply:
F2Fing, the existing code does exactly what I want.  Sorry.

Reply:
like the first loop - I used slice of pointers and keep just the right references in the new slice that we process.